src/runtime_src/core/tools/common/ReportBOStats.cpp

```diff
@@ -73,40 +70,50 @@ ReportBOStats::getPropertyTree20202( const xrt_core::device * pDevice,
 
     //Skip DDR usage info by looking for line with string "BO Stats Below"
     //Following that are BO stats. BO type, size in KB and num of BOs fields separated by blank space
-    for (auto& line: mem_stat) {
+    for (const auto& line: mem_stat) {
       if (line.empty())
         continue;
 
-      if (found) {
-        //Capture BOStat fields separated by blank space
-        std::vector<std::string> bo_info;
-        boost::split(bo_info, line, boost::is_any_of("\t "));
-        //Expected fileds are BO type, size in KB and Num of BOs
-        if (bo_info.size() != 3)
-          throw xrt_core::error((boost::format("ERROR: Unexpected format in BO Stats. Line: %s") % line).str());
-        
-        boost::property_tree::ptree bo_pt;
-        bo_pt.put("type", bo_info[0].substr(1, bo_info[0].length()-2));
-        bo_pt.put("mem_used", bo_info[1].substr(0, bo_info[1].length()-2));
-        bo_pt.put("mem_used_unit", bo_info[1].substr(bo_info[1].length()-2, 2));
-        bo_pt.put("count", bo_info[2].substr(0, bo_info[2].length()-3));
-        //pt1.add_child("bo", bo_pt);
-        pt2_list.push_back(std::make_pair("", bo_pt));
-        continue;
-      }
-
       //Skip DDR usage stats at begining by looking for beloe string
-      if (line.find("BO Stats Below") != std::string::npos)
+      if (!found && line.find("BO Stats Below") != std::string::npos) {
         found = true;
+	continue;
+      } else if (!found)
+	continue;
+
+      //Capture BOStat fields separated by blank space
+      std::vector<std::string> bo_info;
+      boost::split(bo_info, line, boost::is_any_of("\t "));
+      //Expected fileds are BO type, size in KB and Num of BOs
+      if (bo_info.size() != 3)
+        throw std::runtime_error((boost::format("ERROR: Unexpected format in BO Stats. Line: %s") % line).str());

```

Comment:
is there a reason we switched to runtime_error from xrt_core::error?

Reply:
xbutil2 code is mostly using runtime_error. The change actually was to avoid operation_cancel so that scripts can continue on after an error.